tests/cnf/core/network/metallb/tests/bgp-tests.go

```diff
@@ -172,33 +115,244 @@ var _ = Describe("BGP", Ordered, Label(tsparams.LabelBGPTestCases), ContinueOnFa
+		WithLocalPref(100).
+		Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create BGPAdvertisement")
+
+	By("Deploy nginx on single worker with LB service")
+	setupNGNXPod("nginxpod1", workerNodeList[0].Object.Name, tsparams.LabelValue1)
+	setupMetalLbService(tsparams.MetallbServiceName, netparam.IPV4Family, tsparams.LabelValue1, ipAddressPool,
+		corev1.ServiceExternalTrafficPolicyTypeCluster)
+
+	By("Creating configMap with selected worker nodes as BGP Peers for external FRR Pod")
+
+	var masterConfigMap *configmap.Builder
+
+	if extFrrAdv {
+		masterConfigMap = createConfigMapWithNetwork(
+			tsparams.LocalBGPASN, ipv4NodeAddrList, tsparams.ExtFrrConnectedPool, false, false)
+	} else {
+		masterConfigMap = createConfigMap(tsparams.LocalBGPASN, ipv4NodeAddrList, false, false)
+	}
+
+	By("Creating macvlan NAD for external FRR Pod")
+
+	err = define.CreateExternalNad(APIClient, frrconfig.ExternalMacVlanNADName, tsparams.TestNamespaceName)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a macvlan NAD")
+
+	By("Creating external FRR Pod with configMap mount and external NAD")
+
+	extFrrPod := createFrrPod(masterNodeList[0].Object.Name, masterConfigMap.Object.Name, []string{},
+		pod.StaticIPAnnotation(frrconfig.ExternalMacVlanNADName, []string{fmt.Sprintf("%s/%s", ipv4metalLbIPList[0], "24")}))
+
+	By("Checking that BGP session is established on external FRR Pod")
+	verifyMetalLbBGPSessionsAreUPOnFrrPod(extFrrPod, ipv4NodeAddrList)
+
+	return frrk8sPods, extFrrPod, bgpAdvertisement
+}
+
+func validateFrrk8sPodLogLevel(frrk8spods []*pod.Builder, logLevel string) {
+	for _, frrk8sPod := range frrk8spods {
+		res, err := frrk8sPod.ExecCommand([]string{"vtysh", "-c", "show logging"}, "frr")
+		Expect(err).ToNot(HaveOccurred(), "Failed to get frr log")
+		Expect(res.String()).To(ContainSubstring(logLevel), "Logs does not match the log level")
+	}
+}
+
+func verifyBGPTimer(frrPod *pod.Builder, peerAddrList []string, hTimer, aTimer int) {
+	for _, peerAddress := range netcmd.RemovePrefixFromIPList(peerAddrList) {
+		Eventually(frr.GetBGPNeighborTimer,
+			time.Minute*3, tsparams.DefaultRetryInterval).
+			WithArguments(frrPod, peerAddress, hTimer, aTimer).Should(
+			BeTrue(), "Failed to verify BGP Timer on peer")
+	}
+}
+
+func createConfigMapWithNetwork(
+	bgpAsn int,
+	nodeAddrList, externalAdvertisedIPv4Routes []string,
+	enableMultiHop, enableBFD bool) *configmap.Builder {
+	frrBFDConfig := frr.DefineBGPConfigWithNetwork(bgpAsn, tsparams.LocalBGPASN,
+		externalAdvertisedIPv4Routes, netcmd.RemovePrefixFromIPList(nodeAddrList), enableMultiHop, enableBFD)
+	configMapData := frrconfig.DefineBaseConfig(frrconfig.DaemonsFile, frrBFDConfig, "")
+	masterConfigMap, err := configmap.NewBuilder(APIClient, "frr-master-node-config", tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")
+
+	return masterConfigMap
+}
```

Comment:
removed enableMultiHop, enableBFD bool
renamed frrBFDConfig
```suggestion
func createConfigMapWithNetwork(
	bgpAsn int,
	nodeAddrList, externalAdvertisedIPv4Routes []string,) *configmap.Builder {
	frrBGPConfig := frr.DefineBGPConfigWithNetwork(bgpAsn, tsparams.LocalBGPASN,
		externalAdvertisedIPv4Routes, netcmd.RemovePrefixFromIPList(nodeAddrList), false, false)
	configMapData := frrconfig.DefineBaseConfig(frrconfig.DaemonsFile, frrBFDConfig, "")
	masterConfigMap, err := configmap.NewBuilder(APIClient, "frr-master-node-config", tsparams.TestNamespaceName).
		WithData(configMapData).Create()
	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")

	return masterConfigMap
}
```